added sensitive data scanner support and initial lambda script

alinvetian opened the pull request:
Closes: https://github.com/datadryad/dryad-product-roadmap/issues/3410

alinvetian commented:
Tested from dev server:

Rails console input/output:

```
[8] pry(main)> StashEngine::DataFile.last.trigger_sensitive_data_scan
=> {:triggered=>true, :msg=>""}
[9] pry(main)> StashEngine::DataFile.last.pii_scan_report
=> #<StashEngine::PiiScanReport:0x00007f0e1b6636d8
 id: 1,
 report: "{\"report\": {\"errors\": [\"File type not supported\"], \"valid\": true, \"issues\": []}}",
 generic_file_id: 15739,
 status: "noissues",
 created_at: Tue, 22 Oct 2024 16:43:50.794666000 UTC +00:00,
 updated_at: Tue, 22 Oct 2024 16:43:50.794666000 UTC +00:00>
```

CloudWatch output:
```
  | 2024-10-22T19:46:02.265+03:00 | START RequestId: f8f43764-87b4-49bc-b2b8-960b0c443930 Version: $LATEST
  | 2024-10-22T19:46:02.265+03:00 | d14fdd0731203d47add2c27b81ea0340 - parsing file: https://dryad-s3-dev.s3.us-west-2.amazonaws.com/ce085816-4223/data/README.md
  | 2024-10-22T19:46:02.265+03:00 | d14fdd0731203d47add2c27b81ea0340 - callback_url: https://v3-dev.datadryad.org/api/v2/files/15739/piiScanReport
  | 2024-10-22T19:46:02.265+03:00 | d14fdd0731203d47add2c27b81ea0340 - end parsing with status: noissues
  | 2024-10-22T19:46:02.415+03:00 | END RequestId: f8f43764-87b4-49bc-b2b8-960b0c443930
  | 2024-10-22T19:46:02.415+03:00 | REPORT RequestId: f8f43764-87b4-49bc-b2b8-960b0c443930 Duration: 150.25 ms Billed Duration: 151 ms Memory Size: 128 MB Max Memory Used: 59 MB
```

ryscher commented:
Overall, looks good. 

A few minor things to adjust:
- The "coordinates" scanner is too loose -- it will match almost anything in a CSV file. Please remove it.
- It needs cutoffs to ensure it works properly on larger files - when I submitted a file with 2000 issues, the report did not make it back to Dryad.
- The names should be parallel. The name to run the report (`trigger_sensitive_data_scan`) should match the name to view the report (`pii_scan_report`)

alinvetian commented:
@ryscher I updated and addressed all 3 comments.
I added `CUTOFF_ERROR_COUNT = 20` and also returned in the report the `run_time` in seconds and `finished` flag so we know if the report finished parsing or not

All changes are uploaded on AWS Lambda
